### PR TITLE
Add the apps group.

### PR DESCRIPTION
**Pull request by oponder**
towards: https://github.com/giantswarm/giantswarm/issues/4331

This adds the `app.giantswarm.io` api group, which being discussed here: https://docs.google.com/document/d/1D2lQveGYbt2qd8nVJ8tW_0kCs6V0-62NrVxZhcyXPrY/

### CODE REVIEW

**Review by puja108 — APPROVED** [review]
LGTM from my side, but would like an eye or two from @giantswarm/sig-operator and do we want to also ping @giantswarm/sig-product on apiextensions PRs nowadays?

File: pkg/apis/app/v1alpha1/register.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	group   = "app.giantswarm.io"
+	version = "v1alpha1"
```
Comment (kopiczko) [inline pkg/apis/app/v1alpha1/register.go]: Just saying: at some point we should release a stable version if this is a user-facing CRD.

**Review by kopiczko — APPROVED** [review]
LGTM

**Comment by oponder** [comment]
Ok, I leave this PR open for now until I start to do something that actually implements it and see it running and working.

**Comment by oponder** [comment]
closing this as it's gone further thx to @rossf7 and @tomahawk28

**Pull request closed without merging**